assignments/lambda-classes.js

```diff
@@ -1 +1,182 @@
+    this.className = attributes.className;
+    this.favSubjects = attributes.favSubjects;
+    //Stretch
+    this.grade = attributes.grade;
+  }
+
+  listsSubjects() {
+    this.favSubjects.forEach(element => console.log(element));
+  }
+
+  PRAssignment(subject) {
+    console.log(`${this.name} has submitted a PR for ${subject}`);
+  }
+
+  sprintChallenge(subject) {
+    console.log(`${this.name} has begun a spring challenge for ${subject}`);
+  }
+
+  //Stretch
+  graduate() {
+    if (this.grade > 70) {
+      console.log(`Congratulations ${this.name}! You've graduated!`);
+    } else {
+      console.log(`Whoops! You're not ready to graduate yet! Back to work!`);
+    }
+  }
+}
+
+class ProjectManager extends Instructor {
+  constructor(attributes) {
+    super(attributes);
+    this.gradClassName = attributes.gradClassName;
+    this.favInstructor = attributes.favInstructor;
+  }
+
+  standUp(slack) {
+    console.log(`${this.name} announces to ${slack}, @channel standy times!`);
+  }
+
+  debugCode(student, subject) {
+    console.log(`${this.name} debugs ${student.name}'s code on ${subject}`);
+  }
+}
+
+//Objects using the above classes
+const fred = new Instructor({
+  name: 'Fred',
+  location: 'Bedrock',
+  age: 37,
+  favLanguage: 'JavaScript',
+  specialty: 'Front-end',
+  catchPhrase: `Don't forget the homies`,
+});
+
+const dan = new Instructor({
+  name: 'Dan',
+  age: Infinity,
+  location: 'Denver',
+  specialty: 'Relentless Debugger',
+  favLanguage:
+    'JavaScript, Python, Elm, ReasonML, SmallTalk, Haskell, C#, Java, Rust, Go, Ruby, Crystal, Elixir, Lua, and Julia',
+  catchPhrase: 'If you can do the thing, you can get paid to do the thing!',
+});
+
+const brandon = new Student({
+  name: 'Brandon',
+  age: 26,
+  location: 'California',
+  previousBackground: 'Retail',
+  className: 'WEB21',
+  favSubjects: ['HTML', 'CSS', 'Javascript'],
+  //Stretch
+  grade: 70,
+});
+
+const isaiah = new Student({
+  name: 'Isaiah',
+  age: 18,
+  location: 'Florida',
+  previousBackground: 'High School last month',
+  className: 'Web21',
+  favSubjects: ['Html', 'CSS', 'JavaScript'],
+  //Stretch
+  grade: 70,
+});
+
+const marguel = new ProjectManager({
+  name: 'Marguel',
+  age: 'Maybe 26',
+  gradClassName: 'WEBPT2',
+  favInstructor: 'Me?',
+  location: 'California',
+  specialty: 'React',
+  favLanguage: 'JavaScript, Python, Elm etc.',
+  catchPhrase: 'Practice Flex Zombies !!!',
+});
+
+const mary = new ProjectManager({
+  name: 'Mary',
+  age: '24',
+  gradClassName: 'WEB18',
+  favInstructor: 'Josh Knell',
+  location: 'New York',
+  specialty: 'Express and Node.js',
+  favLanguage: 'Javascript',
+  catchPhrase: 'That looks AWESOME',
+});
+
+//Using above objects in examples
+
+//Instructors
+console.log(fred.name);
+fred.speak();
+fred.demo('Javascript Arrays');
+console.log(dan.specialty);
+dan.grade(brandon, 'Javascript Arrays');
+
+//Students
+console.log(brandon.name);
```

Comment:
With all of these console logs, it would be a *great* place to comment the expected results of each beside each log to get a feel for using test driven development.